### ISSUE TITLE
Panic when calling check_and_set with 0 capacity.

### ISSUE THREAD

**Issue by Craig-Macomber** [dpbriggs/growable-bloom-filters#11]
Panic when calling check_and_set with 0 capacity.

```rust
    #[test]
    fn bloom() {
        let mut filter = GrowableBloom::new(0.05, 0);
        filter.check_and_set(1);
    }
```

This panics in `Bloom::new` due to `debug_assert!(capacity >= 1);`

> assertion failed: capacity >= 1
> stack backtrace:
>    0: rust_begin_unwind
>              at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/std/src/panicking.rs:665:5
>    1: core::panicking::panic_fmt
>              at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:74:14
>    2: core::panicking::panic
>              at /rustc/eeb90cda1969383f56a2637cbd3037bdf598841c/library/core/src/panicking.rs:148:5
>    3: growable_bloom_filter::Bloom::new
>              at /home/craig/.cargo/registry/src/index.crates.io-6f17d22bba15001f/growable-bloom-filter-2.1.0/src/lib.rs:45:9
>    4: growable_bloom_filter::GrowableBloom::grow
>              at /home/craig/.cargo/registry/src/index.crates.io-6f17d22bba15001f/growable-bloom-filter-2.1.0/src/lib.rs:506:25
>    5: growable_bloom_filter::GrowableBloom::insert
>              at /home/craig/.cargo/registry/src/index.crates.io-6f17d22bba15001f/growable-bloom-filter-2.1.0/src/lib.rs:388:13
>    6: growable_bloom_filter::GrowableBloom::check_and_set
>              at /home/craig/.cargo/registry/src/index.crates.io-6f17d22bba15001f/growable-bloom-filter-2.1.0/src/lib.rs:488:10

I think this is a bug: I discovered it since my app hits it when populating the bloom filter got an empty cache on startup (set initial capacity estimate to size of cache, which is zero in that edge case).

If zero is really invalid, GrowableBloom::bloom's documentation should probably be a bit more clear about it. I assumed that since its "grow-able" setting the est_insertions is mainly a performance things (like vector capacity) and not critical for correctness.

I've worked around this by being smarter with my est_insertions value, but I figured this should probably be reported.

**Comment by dpbriggs** [comment]
Thanks for the report! I did a quick skim and I would need to refactor a decent portion of the internals to support zero (or very small) capacity. In the meanwhile I updated the documentation to specify that capacity must not be non-zero.

Ideally the interface would return a Result but that would be a breaking change.

**Comment by dpbriggs** [comment]
I've published [2.1.1](https://crates.io/crates/growable-bloom-filter/2.1.1) with the updated documentation
